### PR TITLE
CDAP-5120 Hide local datasets from list call.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://issues.cask.co/browse/CDAP-5120

### CODE REVIEW

**Comment by sagarkapare** [comment]
Build is running here - http://builds.cask.co/browse/CDAP-DUT4011-1

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java [hunk 1]
```diff
@@ -475,13 +476,21 @@ private void executeCondition(ApplicationSpecification appSpec, WorkflowConditio
     executeAll(iterator, appSpec, instantiator, classLoader, token);
   }
 
+  private DatasetProperties addLocalDatasetProperty(DatasetProperties properties) {
+    String dsDescription = properties.getDescription();
+    Map<String, String> dsProperties = properties.getProperties();
+    dsProperties.put(Constants.AppFabric.WORKFLOW_LOCAL_DATASET_PROPERTY, "true");
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java]: isn't this unmodifiable?  we can do `addAll(properties.getProperties).add(Constants.AppFabric.WOR...)` with the builder instead.

**Comment by albertshau** [comment]
one comment

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java (hunk 1 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java]: aah.. good catch 👍 fixed.

**Comment by sagarkapare** [comment]
@albertshau addressed comments. Please look again. Thanks!

**Comment by albertshau** [comment]
lgtm

**Comment by sagarkapare** [comment]
Build is running here - http://builds.cask.co/browse/CDAP-DUT4011-2

**Comment by sagarkapare** [comment]
Build is green here - http://builds.cask.co/browse/CDAP-DUT4011-2. Merging.

**Pull request merged**